include/arch/arm/cortex_m/exc.h

```diff
@@ -27,8 +27,7 @@ extern "C" {
 #define _ZERO_LATENCY_IRQS_RESERVED_PRIO 0
 #endif
 
-#if defined(CONFIG_CPU_CORTEX_M_HAS_PROGRAMMABLE_FAULT_PRIOS) || \
-	defined(CONFIG_CPU_CORTEX_M_HAS_BASEPRI)
+#if defined(CONFIG_CPU_CORTEX_M_HAS_PROGRAMMABLE_FAULT_PRIOS)
```

Comment:
It's cleaner to just do

#define _EXCEPTION_RESERVED_PRIO IS_ENABLED(CONFIG_CPU_CORTEX_M_HAS_PROGRAMMABLE_FAULT_PRIOS)

or better yet get rid of _EXCEPTION_RESERVED_PRIO if possible.

Reply:
Yeah, I think more cleanup, here would be nice

Reply:
However, this proposal is not easy to maintain, since AFAIK IS_ENABLED() is boolean, while _EXCEPTION_RESERVED_PRIO could, in general, be any integer number